src/main/resources/net/sf/cdk/tools/doclets/cheminf.bibx

```diff
@@ -1,11 +1,45 @@
-<?xml version="1.0" encoding="iso-8859-1"?>
+<?xml version="1.0" encoding="utf-8"?>
 <bibtex:file xmlns:bibtex="http://bibtexml.sf.net/"
   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
   xsi:schemaLocation="http://bibtexml.sf.net/ bibtexml.xsd">
 
   <!-- This file is preferably edited in jEdit with the XML plugin installed -->
 
-  <bibtex:entry id="AiresDeSousa2002">
+  <bibtex:entry bibtex:id="Murray1996">
```

Comment:
Interesting... according to XML, the attribute follows the namespace of the element, unless overwritten (IIRC).

Reply:
https://www.w3.org/TR/REC-xml-names/#scoping-defaulting specifies:

> Default namespace declarations do not apply directly to attribute names; the interpretation of unprefixed attributes is determined by the element on which they appear.

and then immediately below:

> The namespace name for an unprefixed attribute name always has no value.

from which I infer that a "bare" id= attribute does *not* get the default namespace.

Then there is https://www.w3.org/TR/xmlschema11-1/ that says:
> Control over whether attribute information items ·validated· by a local declaration must be similarly qualified or not is provided by the form [attribute], whose default is provided by the attributeFormDefault [attribute] on the enclosing <schema>, via its determination of {target namespace}.

From this (admittedly quite obscure) passage I understand that schema definition controls whether the 'id' attribute will be qualified or unqualified; defining it as '<xsd:attribute name="id" form="unqualified" ...' or specify attributeFormDefault="unqualified" in the <xsd:schema>' element will require the id elements to be *unqialified*; specifying either form= or attributeFormDefault= as "qualified" will require the 'id' attribute to be always qualified, regardless of the default namespace setting. The unqualified 'id' attributed will have no namespace (and belong to a sort of "global namespace"?).

The bibtexml-handwritten.xsd schema currently specifies 'attributeFormDefault="qualified"'. The cheminf.bibx specifies unqualified id ('<bibtex:entry id="AiresDeSousa2002">). Thus, it does not validate. The behaviour of the 'xmllint' tool ('sudo ap install xmlllint' on Ubuntu-16.04) is consistent with above considerations.

To get cheminf.bibx validated, one would need either to change 'attributeFormDefault="unqualified"' in bibtexml*.xsd, or add the 'bibtex:' namespace to the 'id' element ('<bibtex:entry bibtex:id="AiresDeSousa2002">). I've tested both methods with xmllint, they work.

Reply:
Thanks, that settles it... I know this was one of the harder parts of XML and early parsers did not all agree... so much, that I had remembered it wrong. Thanks for looking up and listing here the details!